presto-raptorx/src/main/java/com/facebook/presto/raptorx/metadata/DatabaseMetadataWriter.java

```diff
@@ -0,0 +1,728 @@
+            checkConflict(dao.tableIdExists(tableId), "Table no longer exists");
+
+            table.set(dao.getTableInfo(tableId).builder()
+                    .addRowCount(chunks.stream().mapToLong(ChunkInfo::getRowCount).sum())
+                    .setUpdateTime(currentTime)
+                    .build());
+
+            columns.set(dao.getColumnInfo(tableId));
+            Set<Long> tableColumnIds = columns.get().stream()
+                    .map(ColumnInfo::getColumnId)
+                    .collect(toSet());
+            checkConflict(tableColumnIds.containsAll(columnIds), "Column no longer exists");
+
+            deleteTable(dao, commitId, tableId);
+            insertTable(dao, commitId, table.get());
+
+            dao.updateRollbackInfo(commit.getRollbackInfo()
+                    .withWrittenTable(tableId)
+                    .toBytes());
+        });
+
+        for (Collection<ChunkInfo> batch : partition(chunks, 10000)) {
+            writeShard(commitId, getTableShard(tableId), dao -> {
+                dao.insertChunks(commitId, tableId, currentTime, batch.stream()
+                        .map(chunk -> ChunkMetadata.from(table.get(), chunk))
+                        .iterator());
+
+                try (IndexWriter writer = new IndexWriter(dao.getHandle().getConnection(), commitId, tableId, columns.get())) {
+                    for (ChunkInfo chunk : batch) {
+                        writer.add(chunk.getChunkId(), chunk.getBucketNumber(), chunk.getColumnStats());
+                    }
+                    writer.execute();
+                }
+                catch (SQLException e) {
+                    throw metadataError(e);
+                }
+            });
+        }
+
+        writeShard(commitId, getTableShard(tableId), dao -> {
+            TableSize tableSize = dao.getTableSize(tableId);
+            long chunkCount = tableSize.getChunkCount();
+            long compressedSize = tableSize.getCompressedSize();
+            long uncompressedSize = tableSize.getUncompressedSize();
+
+            chunkCount += chunks.size();
+            compressedSize += chunks.stream().mapToLong(ChunkInfo::getCompressedSize).sum();
+            uncompressedSize += chunks.stream().mapToLong(ChunkInfo::getUncompressedSize).sum();
+
+            deleteTableSize(dao, commitId, tableId);
+            dao.insertTableSize(commitId, tableId, chunkCount, compressedSize, uncompressedSize);
+        });
+    }
+
+    @Override
+    public void deleteChunks(long commitId, long tableId, Set<Long> chunkIds)
+    {
+        writeMaster(commitId, (dao, commit) -> {
+            checkConflict(dao.tableIdExists(tableId), "Table no longer exists");
+            dao.updateRollbackInfo(commit.getRollbackInfo()
+                    .withWrittenTable(tableId)
+                    .toBytes());
+        });
+
+        AtomicLong rowCount = new AtomicLong();
+
+        writeShard(commitId, getTableShard(tableId), dao -> {
```

Comment:
Will here suffer the same "mysql commit size" as in insert_chunks? Or deleteChunks has smaller commit size?

Reply:
I'll break this up.